### PR TITLE
Disallow nested amp-list in AMP4EMAIL

### PR DESCRIPTION
**Pull request by choumx**
Enforce this rule at runtime for script templates: https://github.com/ampproject/amphtml/blob/88c71f02112e6d1b62e173c8ea4396c25e0dba2f/extensions/amp-list/validator-amp-list.protoascii#L181-L187

### CODE REVIEW

File: src/sanitation.js [hunk 1]
```diff
@@ -247,18 +280,22 @@ export function isValidAttr(
   }
 
   // Remove blacklisted attributes from specific tags e.g. input[formaction].
-  const attrBlacklist = isAmp4Email(doc)
-    ? EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName]
-    : BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName];
+  const attrBlacklist = Object.assign(
+    map(),
+    BLACKLISTED_TAG_SPECIFIC_ATTRS,
+    isAmp4Email(doc) ? EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTRS : {}
```
Comment (alabiaga) [inline src/sanitation.js]: nit can we save a result of the isAmp4Email(doc) call to be used here and below.

File: src/sanitation.js [hunk 2]
```diff
@@ -247,18 +280,22 @@ export function isValidAttr(
   }
 
   // Remove blacklisted attributes from specific tags e.g. input[formaction].
-  const attrBlacklist = isAmp4Email(doc)
-    ? EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName]
-    : BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName];
+  const attrBlacklist = Object.assign(
```
Comment (alabiaga) [inline src/sanitation.js]: can we create a private blacklistedTags_ call which the blacklistedTags will call. Use private call here and below

e.g.

const blacklistedTags_ = getBlacklistedTags_.bind(null, isAmp4Email(doc));

export function blacklistedTags(doc) {
  return  blacklistedTags_(BLACKLISTED_TAGS, EMAIL_BLACKLISTED_TAGS);
}

function getBlacklistedTags_(isAmp4Email, blacklistTarget, blacklistSource) {
  return Object.assign(map(), blacklistTarget, isAmp4Email ? blacklistSource {});
}

const attrBlacklist = blacklistTags_(BLACKLISTED_TAG_SPECIFIC_ATTRS, 
    EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTRS);

const attrValueBlacklist = blacklistTags_(
    BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES,
    EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES)[tagName];

Comment (choumx) [reply src/sanitation.js]: Is the intent to have code reuse for the `Object.assign` call? It seems harder to read IMO.

Comment (alabiaga) [reply src/sanitation.js]: Yes, I don't feel strongly about it so that's fine. I'll look over the rest of the changes before end of the day. Thanks

**Pull request merged**